src/target/nrf91.c

```diff
@@ -0,0 +1,106 @@
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+#include "adiv5.h"
+
+/* Non-Volatile Memory Controller (NVMC) Registers */
+#define NRF91_NVMC          0x50039000U
+#define NRF91_NVMC_READY    (NRF91_NVMC + 0x400U)
+#define NRF91_NVMC_CONFIG   (NRF91_NVMC + 0x504U)
+#define NRF91_NVMC_ERASEALL (NRF91_NVMC + 0x50cU)
+
+#define NRF91_NVMC_CONFIG_REN  0x0U // Read only access
+#define NRF91_NVMC_CONFIG_WEN  0x1U // Write enable
+#define NRF91_NVMC_CONFIG_EEN  0x2U // Erase enable
+#define NRF91_NVMC_CONFIG_PEEN 0x3U // Partial erase enable
+
+static bool nrf91_wait_ready(target_s *const t, platform_timeout_s *const timeout)
+{
+	/* Poll for NVMC_READY */
+	while (target_mem_read32(t, NRF91_NVMC_READY) == 0) {
+		if (target_check_error(t))
+			return false;
+		if (timeout)
+			target_print_progress(timeout);
+	}
+	return true;
+}
+
+static bool nrf91_flash_erase(target_flash_s *f, target_addr_t addr, size_t len)
+{
+	target_s *t = f->t;
+
+	/* Enable erase */
+	target_mem_write32(t, NRF91_NVMC_CONFIG, NRF91_NVMC_CONFIG_EEN);
+	if (!nrf91_wait_ready(t, NULL))
+		return false;
+
+	for (size_t offset = 0; offset < len; offset += f->blocksize) {
+		/* Write all ones to first word in page to erase it */
+		target_mem_write32(t, addr + offset, 0xffffffffU);
+
+		if (!nrf91_wait_ready(t, NULL))
+			return false;
+	}
+
+	/* Return to read-only */
+	target_mem_write32(t, NRF91_NVMC_CONFIG, NRF91_NVMC_CONFIG_REN);
+	return nrf91_wait_ready(t, NULL);
+}
+
+static bool nrf91_flash_write(target_flash_s *f, target_addr_t dest, const void *src, size_t len)
+{
+	target_s *t = f->t;
+
+	/* Enable write */
+	target_mem_write32(t, NRF91_NVMC_CONFIG, NRF91_NVMC_CONFIG_WEN);
+	if (!nrf91_wait_ready(t, NULL))
+		return false;
+	/* Write the data */
+	target_mem_write(t, dest, src, len);
+	if (!nrf91_wait_ready(t, NULL))
+		return false;
+	/* Return to read-only */
+	target_mem_write32(t, NRF91_NVMC_CONFIG, NRF91_NVMC_CONFIG_REN);
+	return true;
+}
+
+static void nrf91_add_flash(target_s *t, uint32_t addr, size_t length, size_t erasesize)
+{
+	target_flash_s *f = calloc(1, sizeof(*f));
+	if (!f) { /* calloc failed: heap exhaustion */
+		DEBUG_WARN("calloc: failed in %s\n", __func__);
+		return;
+	}
+
+	f->start = addr;
+	f->length = length;
+	f->blocksize = erasesize;
+	f->erase = nrf91_flash_erase;
+	f->write = nrf91_flash_write;
+	f->erased = 0xff;
+	target_add_flash(t, f);
+}
+
+bool nrf91_probe(target_s *t)
```

Comment:
As this is all new code, and we're trying to reduce the amount of single letter variable names (it is a clang-tidy lint for a reason) could `t` be renamed `target` please. Similarly above and for `f` to be renamed `flash`